python/tank/context.py

```diff
@@ -668,6 +668,83 @@ def create_empty(tk):
     """
     return Context(tk)
 
+def from_entities(tk, project, entity, step, task, user):
+    """
+    Construct a new context from the specified entities.
+
+    Note: It is the responsibility of the calling code to ensure that both the entities and the
+    relationships between them are valid.  
+    
+    This only does a Shotgun query if the context_additional_entities hook specifies additional
+    fields on a task to look for additional entities on!
+
+    :param tk:      An sgtk API instance
+    :param project: The project entity dictionary to use when constructing the context
+    :param entity:  The entity entity dictionary to use when constructing the context
+    :param step:    The step entity dictionary to use when constructing the context
+    :param task:    The task entity dictionary to use when constructing the context
+    :param user:    The user entity dictionary to use when constructing the context
+    :returns:       A Context constructed from the specified entities.
+    """
+    # prep our return data structure
+    context = {
+        "tk": tk,
+        "project": None,
+        "entity": None,
+        "step": None,
+        "user": None,
+        "task": None,
+        "additional_entities": []
+    }    
+
+    if project and "type" in project and "id" in project:
+        context["project"] = {
+            "type":project["type"], 
+            "id":project["id"], 
+            "name":project.get("name")
```

Comment:
There is actually a bug ticket queued up for either this or next sprint where the context doesn't have name already, which path cache is assuming.  Should be consistent here, but I don't think we can assume name is always there, since the Shotgun API is inconsistent itself about when that field is automatically there.